### PR TITLE
sratookit 2.4.3

### LINKED ISSUE #1371: sratoolkit: Update?
NCBI changed the whole structure for sratoolkit repos splitting it into three separate repos[1,2,3]
`sra-tools` requires the other two `ngs` and `ncbi-vdb` as buildtime  dependency. 

The current formula https://github.com/Homebrew/homebrew-science/blob/master/sratoolkit.rb relies on installing Mac specific binaries and hence does not work on linux.

NCBI also distributes pre-built binaries[4] for different platforms. A formula update has two possible directions:

1. Discard the binary approach and write formulae for all 3 separate repos
2. Update the current binary install approach for different platforms

I have a partial implementation for (1) but I am not sure if that is an overkill since `ngs` and `ncbi-vdb` are required only during build time and I have not come across standalone usecases. Note `ngs` and `ncbi-vdb` are tagged too(https://github.com/ncbi/ngs/releases, https://github.com/ncbi/ncbi-vdb/releases)

What is the best approach in such cases? Is it okay to write a custom method inside the current formula that will download these two dependencies rather than making 2 additional new formulae?

[1] https://github.com/ncbi/sra-tools
[2] https://github.com/ncbi/ncbi-vdb
[3] https://github.com/ncbi/ngs
[4] https://github.com/ncbi/sra-tools/wiki/Downloads

cc @sjackman

### PR DESCRIPTION
**Pull request by saketkc**
Fixes #1371

### CODE REVIEW

**Comment by saketkc** [comment]
Interesting.
It builds fine here: https://travis-ci.org/saketkc/brewer/builds/46818570#L369

**Comment by sjackman** [comment]
```
==> Downloading https://github.com/ncbi/ncbi-vdb/archive/.tar.gz
Error: Failed to download resource "sratoolkit--ncbi-vdb"
Download failed: https://github.com/ncbi/ncbi-vdb/archive/.tar.gz
```

**Comment by saketkc** [comment]
Fixed.

**Comment by sjackman** [comment]
```
==> make CC=gcc-4.9
gcc-4.9 -o adler32.pic.o -fPIC -O3 -Wall -Wno-variadic-macros -fno-strict-aliasing -D_LIBRARY /private/tmp/sratoolkit--ncbi-vdb-70qMux/ncbi-vdb-2.4.3/libs/ext/zlib/adler32.c -D__mod_name__=libs/ext/zlib -D__file_name__=adler32 -D__file_ext__=c
Undefined symbols for architecture x86_64:
  "_main", referenced from:
     implicit entry/start for main executable
ld: symbol(s) not found for architecture x86_64
collect2: error: ld returned 1 exit status
make[4]: *** [adler32.pic.o] Error 1
make[3]: *** [std] Error 2
make[2]: *** [zlib] Error 2
make[1]: *** [ext] Error 2
make: *** [libs] Error 2
```

**Comment by sjackman** [comment]
I encourage your getting this to build from source. In the mean time, here's a binary build.
https://github.com/Homebrew/homebrew-science/pull/1713

**Comment by sjackman** [comment]
Also https://github.com/Homebrew/homebrew-science/pull/1714

**Comment by iml** [comment]
This builds fine with clang on my machine once the reference to `libmagic-static` is fixed. (e.g. `inreplace "tools/copycat/Makefile", "smagic-static", "smagic"`)

However, it conflicts with shogun and there seems to be a lot of cruft from the build leftover in prefix.

File: sratoolkit.rb [hunk 1]
```diff
@@ -1,34 +1,52 @@
...
+    end
+
+    system "./configure", "--prefix=#{prefix}", "--with-ngs-sdk-prefix=#{prefix}", "--with-ncbi-vdb-sources=#{include}/ncbi-vdb", "--with-ncbi-vdb-build=#{prefix}", "--prefix=#{prefix}", "--build=#{prefix}"
+    system "make"
+    system "make", "install"
   end
 
   test do
-    system "#{bin}/fastq-dump", "--version"
+    system "fastq-dump", "--version"
```
Comment (sjackman) [inline sratoolkit.rb]: Revert this change. Test should use an absolute prefix so testing of unlinked formulas work.

**Comment by sjackman** [comment]
```
clang -Wl,-arch,x86_64 -o /usr/local/Cellar/sratoolkit/2.4.3/sra-tools/mac/clang/x86_64/rel/bin/copycat.2.4.3 -DNDEBUG -m64 -mmacosx-version-min=10.6 copycat.o ccfileformat.o cccat.o cctree.o cctree-dump.o cctar.o ccsra.o ccsubchunk.o ccfile.o -L/usr/local/Cellar/sratoolkit/2.4.3/sra-tools/mac/clang/x86_64/rel/lib -L/usr/local/Cellar/sratoolkit/2.4.3/sra-tools/mac/clang/x86_64/rel/ilib -L/usr/local/Cellar/sratoolkit/2.4.3/lib64 -L/usr/local/Cellar/sratoolkit/2.4.3/ncbi-vdb/mac/clang/x86_64/rel/lib -L/usr/local/Cellar/sratoolkit/2.4.3/ncbi-vdb/mac/clang/x86_64/rel/ilib -lkapp -lncbi-vdb-static -lkff -lmagic-static -ldl -lpthread -lxml2 -lm
ld: library not found for -lmagic-static
collect2: error: ld returned 1 exit status
make[3]: *** [/usr/local/Cellar/sratoolkit/2.4.3/sra-tools/mac/clang/x86_64/rel/bin/copycat] Error 1
```

**Comment by saketkc** [comment]
I would not be able to test it out on a Mac. Submitted an issue upstream: https://github.com/ncbi/sra-tools/issues/6

**Comment by sjackman** [comment]
1. Should `sratoolkit` depend on `libmagic`?
2. `libmagic` does not install a library named `magic-static`

```
❯❯❯ brew list libmagic
/usr/local/Cellar/libmagic/5.22/include/magic.h
/usr/local/Cellar/libmagic/5.22/lib/libmagic.1.dylib
/usr/local/Cellar/libmagic/5.22/lib/libmagic.dylib
/usr/local/Cellar/libmagic/5.22/share/man/ (2 files)
/usr/local/Cellar/libmagic/5.22/share/misc/magic.mgc
```

**Comment by saketkc** [comment]
Required for copycat: https://github.com/ncbi/sra-tools/wiki/Building-and-Installing-from-Source

**Comment by saketkc** [comment]
This fails because the travis script is a bit buggy. It compares the head against the parent commit of this PR. Maybe I would have to get that fixed first.

**Comment by saketkc** [comment]
Rebased at https://github.com/Homebrew/homebrew-science/pull/1811

**Comment by sjackman** [comment]
Hi, Saket. You don't need to open a new PR to rebase. You can rebase your feature branch and `git push --force` it to update the PR.

**Comment by saketkc** [comment]
Yes, my merge went awry.
